vector-im/element-android
# Update notifications setting when m.local_notification_settings.<device-id> event changes for current device (PSG-874)

## Type of change


## Content


Modifications to support changes of the account data event for the current session from another session. We listen first for account data changes and then for pusher changes to update the local notifications setting.

## Motivation and context


Closes #7632 

## Screenshots / GIFs





## Tests



Tests should be performed for both FDroid and Gplay version.
Ntfy app should be installed to have alternative notification method.

- Enable new session manager in labs
- Go to "Settings -> Notifications"
- Select background sync as notifications method
- Check you can toggle notification for current session from the current device or from another session (e.g. from web)
- Check the display is correct in both "Settings -> Notifications" and in overview screen of the current session in "Settings -> Security & Privacy -> Show all sessions"
- Go to "Settings -> Notifications"
- Select "Ntfy" or "Google services" as notifications method
- Check you can toggle notification for current session from the current device or from another session (e.g. from web)
- Check the display is correct in both "Settings -> Notifications" and in overview screen of the current session in "Settings -> Security & Privacy -> Show all sessions". Note when toggling from other session refresh is not immediate since a refresh of the pushers is needed to reflect the correct value.

## Tested devices

- OS version(s): Android 11

## Checklist



- x ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
